src/component/handlers/edit/editOnPaste.js

```diff
@@ -124,6 +126,14 @@ function editOnPaste(e: SyntheticClipboardEvent): void {
         );
         return;
       }
+    // Safari does not properly store text/html. Use the internalClipboard
+    // if present and equal to what is on the clipboard.
+    } else if (isSafari && internalClipboard &&
```

Comment:
HTML not being stored because of a bug: https://bugs.webkit.org/show_bug.cgi?id=19893 (you can see what is stored in #147)

I'm happy to move the code wherever, since it was Paste specific I placed it here.